plugins/ssr/ssrequality.ml

```diff
@@ -720,18 +722,18 @@ let unfoldtac occ ko t kt =
 
 let unlocktac ist args =
   let open Proofview.Notations in
-  Proofview.Goal.enter begin fun gl ->
-  let env = Proofview.Goal.env gl in
-  let sigma = Proofview.Goal.sigma gl in
   let utac (occ, gt) =
     Proofview.Goal.enter begin fun gl ->
       let env = Proofview.Goal.env gl in
       let sigma = Proofview.Goal.sigma gl in
       unfoldtac occ occ (interp_term env sigma ist gt) (fst gt)
     end
   in
-  let sigma, locked = mkSsrConst "locked" env sigma in
-  let sigma, key = mkSsrConst "master_key" env sigma in
+  Proofview.Goal.enter_one ~__LOC__ begin fun _ ->
+    Ssrcommon.tacMK_SSR_CONST "locked" end >>= fun locked ->
+    Ssrcommon.tacMK_SSR_CONST "master_key" >>= fun key ->
+  Proofview.Goal.enter begin fun gl ->
+  let sigma = Proofview.Goal.sigma gl in
```

Comment:
Redundant enter, just get the sigma out with `tclEVARMAP`.

Reply:
Actually you don't even need to reset it below.

Reply:
right